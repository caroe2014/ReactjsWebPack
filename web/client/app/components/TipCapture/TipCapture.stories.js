// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import { storiesOf } from '@storybook/react';
import TipCapture from '../TipCapture';

storiesOf('TipCapture', module)
  .add('No Custom Tip', () => {
    const tipDetails = {
      subTotalAmount: 58.20,
      taxAmount: 5.80,
      tips: [15, 20, 25],
      customTipFlag: false
    };
    return (
      <TipCapture tipDetails={tipDetails} showAddButton/>
    );
  }
  )
  .add('Custom Tip', () => {
    const tipDetails = {
      subTotalAmount: 58.20,
      taxAmount: 5.80,
      tipPercent: [15, 20, 25],
      customTipFlag: true
    };
    return (
      <TipCapture tipDetails={tipDetails} showAddButton/>
    );
  }
  );
