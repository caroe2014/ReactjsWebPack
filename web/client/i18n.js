// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import i18next from 'i18next';
import LanguageDetector from 'i18next-browser-languagedetector';
import Backend from 'i18next-locize-backend';

/* istanbul ignore next */
i18next
  .use(LanguageDetector)
  .use(Backend)
  .init({
    // fallback depending on user language
    fallbackLng: {
      'ja': ['ja-JP'],
      'default': ['en-US']
    },
    order: ['querystring', 'navigator'],
    lookupQuerystring: 'lng',
    allowMultiLoading: true,
    backend: {
      projectId: '838d5fce-27b5-4368-8c54-8fcb33577f9a',
      referenceLng: 'en-US',
      version: process.env.ENVIRONMENT === 'production' ? 'latest' : process.env.ENVIRONMENT || process.env.NODE_ENV
    },
    debug: false,
    detection: {
      caches: []
    },
    // have a common namespace used around the full app
    ns: ['core', `domain-${window.location.hostname}`],
    defaultNS: `domain-${window.location.hostname}`,
    fallbackNS: [`domain-${window.location.hostname}`, 'core'],

    keySeparator: false, // we use content as keys

    interpolation: {
      formatSeparator: ','
    },

    react: {
      wait: true
    }
  }, (err, t) => {
    if (err) return console.log('Something went wrong while loading.', err);
  });

export default i18next;
