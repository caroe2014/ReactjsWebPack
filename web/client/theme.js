// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import { css, injectGlobal } from 'styled-components';

const theme = {
  colors: {
    black: '#000000',
    navigationMenu: '#48a942',
    navigationMenuHover: 'cornflowerblue',
    primary: '#347CFF',
    primaryButton: '#347CFF',
    loginField: '#AAA7AC',
    placeHolder: '#909090',
    loginBoxColor: '#434149',
    open: '#48a942',
    close: 'red',
    footer: '#48a942',
    textGrey: '#505050',
    paymentSuccessTitle: '#505050',
    inputBottomBorder: 'lightgray',
    backgroundGrey: '#E6E6E6',
    text: 'black',
    primaryText: '#383838',
    secondaryText: '#a6a6a6',
    hint: '#a6a6a6',
    light: 'white',
    error: '#cc0000',
    tile: '#505050',
    addToCart: '#347CFF',
    itemTile: '#3C474C0',
    darkGrey: '#464646',
    itemBorder: 'lightgrey',
    addOnTab: '#347CFF',
    validationError: '#cc0000',
    loginBackground: '#303030',
    loginFormBackground: '#484848',
    disableBackground: '#E4E4E4',
    addonBg: '#EEEEEE',
    glutenFree: '#8d7356',
    spicy: '#ed1c24',
    vegan: '#0d4a22',
    vegetarian: '#0d4a22',
    new: '#ff6000',
    healthy: '#359919',
    contextBackground: '#f0f0f3',
    loader: '#ababab',
    disableButton: '#D1D1D1'
  },
  fontFamily: 'Roboto',
  receiptFontFamily: 'Roboto',
  fontSize: {
    xl: '2.4rem',
    lg: '1.8rem',
    md: '1.3rem',
    nm: '1rem',
    sm: '0.75rem'
  },
  fontSizeFromProps: (props) => {
    if (props.xl) return props.theme.fontSize.xl;
    if (props.lg) return props.theme.fontSize.lg;
    if (props.md) return props.theme.fontSize.md;
    if (props.nm) return props.theme.fontSize.nm;
    if (props.sm) return props.theme.fontSize.sm;
    return 'inherit';
  },
  spacing: {
    xl: '8em',
    lg: '4em',
    md: '2em',
    nm: '1em',
    sm: '0.5em',
    none: '0em'
  },
  mediaBreakpoints: {
    mobile: (...args) => css`@media (max-width: 32em) {${css(...args)}}`,
    tablet: (...args) => css`@media (max-width: 72em) {${css(...args)}}`,
    desktop: (...args) => css`@media (min-width: 72em) {${css(...args)}}`
  },
  shadow: '#cfcfcf 0.1px 0.1px 3px 0px',
  nav: {
    height: '70px'
  }
};

injectGlobal`
  .agilysys-icon-new {
    color: ${theme.colors.new}
  }
  .agilysys-icon-healthy {
    color: ${theme.colors.healthy}
  }
  .agilysys-icon-vegetarian {
    color: ${theme.colors.vegetarian}
  }
  .agilysys-icon-vegan {
    color: ${theme.colors.vegan}
  }
  .agilysys-icon-spicy {
    color: ${theme.colors.spicy}
  }
  .agilysys-icon-gluten-free {
    color: ${theme.colors.glutenFree}
  }
  @media print {
    @-moz-document url-prefix() {
      .appContainer, .appContainer>div, .appContainer .parent, .appContainer .BottomContainer {
        display: block !important;
      }
    }
  }
`;

export default theme;
