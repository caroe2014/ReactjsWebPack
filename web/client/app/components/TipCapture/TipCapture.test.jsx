// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { ThemeProvider } from 'styled-components';
import theme from 'web/client/theme.js';
import TipCapture from '.';
import jsdom from 'jsdom';
import { createMemoryHistory } from 'history';

import i18n from 'i18next';

configure({
  adapter: new Adapter()
});

/* global describe, it, expect, beforeEach */
/* eslint-disable max-len */
global.window = new jsdom.JSDOM().window;
global.document = window.document;
global.scrollTo = jest.fn();

jest.mock('web/client/i18n', () => ({
  t: (key) => {
    return key;
  }
}));
jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate HoC receive the t function as a prop
  withTranslation: () => Component => {
    Component.defaultProps = { ...Component.defaultProps, t: (key) => key };
    return Component;
  }
}));

jest.mock('web/client/app/reduxpages/ConnectedComponents', () => {
  return {
    ConnectedLoyaltyProcess: <div/>
  };
});

jest.mock('react-i18next', () => ({
  Trans: 'div'
}));

describe('TipCapture', () => {

  let wrapper;
  let mockProps = {
    subTotalAmount: 58.20,
    taxAmount: 5.80,
    tips: [15, 20, 25]
  };
  let mockCurrency = {
    currencyDecimalDigits: '2',
    currencyCultureName: 'en-US',
    currencyCode: 'USD',
    currencySymbol: '$'
  };
  let selectedTipProp = '';
  let total = 1.00;
  let subTotal = 1.00;
  let tax = 1.00;
  const history = createMemoryHistory('/payments');

  beforeEach(() => {
    wrapper = shallow(<TipCapture tipDetails={mockProps} customTipFlag total={total} subTotal={subTotal} tax={tax} currencyDetails={mockCurrency} selectedTipProp={selectedTipProp}/>
    );
  });

  it('does a snapshot test', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render without throwing an error', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('should mount with themes without throwing an error', () => {
    const mountWrapper = mount(<ThemeProvider theme={theme}><TipCapture tipDetails={mockProps} total={total} subTotal={subTotal} tax={tax} currencyDetails={mockCurrency} selectedTipProp={selectedTipProp}/></ThemeProvider>);
    expect(mountWrapper.exists()).toBe(true);
  });

  it('should render Custom Tip when prop is true', () => {
    const mountWrapper = mount(<ThemeProvider theme={theme}><TipCapture tipDetails={mockProps} subTotal={subTotal} tax={tax} customTipFlag currencyDetails={mockCurrency} selectedTipProp={selectedTipProp}/></ThemeProvider>);
    expect(wrapper.find('.custom-tip-button').length).toBe(1);
  });

  describe('when Custom Tip is false', () => {
    it('should not render Image when Image prop is received', () => {
      const TipWrapper = shallow(<TipCapture tipDetails={mockProps} customTipFlag={false} total={total} subTotal={subTotal} tax={tax} currencyDetails={mockCurrency} selectedTipProp={selectedTipProp}/>);
      expect(TipWrapper.find('.custom-tip-button').length).toBe(0);
    });
  });

  describe('When a tip option is clicked', () => {
    it('should update TipOption state', () => {
      expect(wrapper.state().selectedTip).toBe('');
      const input = wrapper.find('.tip-button-0');
      input.simulate('click');
      expect(wrapper.state().selectedTip).toBe(mockProps.tips[0]);
    });
    it('should render theme of TipOption state when selected', () => {
      const tipWrapper = mount(<ThemeProvider theme={theme}><TipCapture tipDetails={mockProps} subTotal={subTotal} tax={tax} customTipFlag currencyDetails={mockCurrency} selectedTipProp={selectedTipProp}/></ThemeProvider>);
      const input = tipWrapper.find('.tip-button-0').at(0);
      input.simulate('click');
      // expect(tipWrapper.state().selectedTip).toBe(mockProps.tips[0]);
      // expect(wrapper.find('.tip-button-0')).toHaveStyleRule('font-size', '16px');
    });
  });

  it('should render No tipbutton always', () => {
    expect(wrapper.dive().find('.no-tip-button').length).toBe(1);
  });

  describe('When CustomTipButton is clicked', () => {
    it('should update selectedTip state', () => {
      expect(wrapper.state().selectedTip).toBe('');
      const input = wrapper.find('.custom-tip-button');
      input.simulate('click');
      expect(wrapper.state().selectedTip).toBe('custom');
    });
  });

  describe('When selectedTipProp is available', () => {
    describe('When selectedTipProp is Custom', () => {
      it('should render', () => {
        const customWrapper = shallow(<TipCapture tipDetails={mockProps} subTotal={subTotal} tax={tax} customTipFlag currencyDetails={mockCurrency} selectedTipProp={'custom'} selectedTipAmountProp={1.00}/>);
        expect(customWrapper.exists()).toBe(true);
      });
      it('should render', () => {
        const customWrapper = mount(<ThemeProvider theme={theme}><TipCapture tipDetails={mockProps} subTotal={subTotal} tax={tax} customTipFlag currencyDetails={mockCurrency} selectedTipProp={'custom'} selectedTipAmountProp={1.00}/></ThemeProvider>);
        expect(customWrapper.exists()).toBe(true);
      });
    });
  });

  describe('When Closebutton is clicked', () => {
    it('should reset customTipValue and customTipInput state', () => {
      const customWrapper = shallow(<TipCapture tipDetails={mockProps} subTotal={subTotal} tax={tax} customTipFlag currencyDetails={mockCurrency} selectedTipProp={'custom'} selectedTipAmountProp={1.00}/>);
      const input = customWrapper.find('.close-button');
      input.simulate('click');
      expect(customWrapper.state().customTipValue).toBe('');
      expect(customWrapper.state().customTipInput).toBe('');
    });

    it('should render with mount', () => {
      const customWrapper = mount(<ThemeProvider theme={theme} ><TipCapture tipDetails={mockProps} subTotal={subTotal} tax={tax} customTipFlag currencyDetails={mockCurrency} selectedTipProp={'custom'} selectedTipAmountProp={1.00}/></ThemeProvider>);
    });
  });

  describe('When props are updated', () => {
    // it('delivery is enabled, it should render', () => {
    //   // const historyMock = { push: jest.fn() };
    //   const updateWrapper = shallow(<TipCapture tipDetails={mockProps} orderConfig deliveryDestination history={history} subTotal={subTotal} tax={tax} customTipFlag currencyDetails={mockCurrency} selectedTipProp={'custom'} selectedTipAmountProp={1.00}/>);
    //   updateWrapper.setProps({ tipFlag: true });
    //   expect(updateWrapper.exists()).toBe(true);
    //   expect(updateWrapper.instance().props.history.location.pathname).toBe('/deliveryLocation');
    // });
    // it('delivery is disabled, it should render', () => {
    //   // const historyMock = { push: jest.fn() };
    //   const updateWrapper = shallow(<TipCapture tipDetails={mockProps} deliveryDestination={false} history={history} subTotal={subTotal} tax={tax} customTipFlag currencyDetails={mockCurrency} selectedTipProp={'custom'} selectedTipAmountProp={1.00}/>);
    //   updateWrapper.setProps({ tipFlag: true });
    //   expect(updateWrapper.exists()).toBe(true);
    //   expect(updateWrapper.instance().props.history.location.pathname).toBe('/deliveryLocation');
    // });
    it('delivery is disabled, name capture is enabled, it should render', () => {
      // const historyMock = { push: jest.fn() };
      const updateWrapper = shallow(<TipCapture tipDetails={mockProps} nameCapture deliveryDestination={false} history={history} subTotal={subTotal} tax={tax} customTipFlag currencyDetails={mockCurrency} selectedTipProp={'custom'} selectedTipAmountProp={1.00}/>);
      updateWrapper.setProps({ tipFlag: true });
      expect(updateWrapper.exists()).toBe(true);
      expect(updateWrapper.instance().props.history.location.pathname).toBe('/nameCapture');
    });
    it('delivery is disabled, name capture is enabled, and Sms is Enabled  it should render', () => {
      // const historyMock = { push: jest.fn() };
      const updateWrapper = shallow(<TipCapture tipDetails={mockProps} isSmsEnabled nameCapture={false} deliveryDestination={false} history={history} subTotal={subTotal} tax={tax} customTipFlag currencyDetails={mockCurrency} selectedTipProp={'custom'} selectedTipAmountProp={1.00}/>);
      updateWrapper.setProps({ tipFlag: true });
      expect(updateWrapper.exists()).toBe(true);
      expect(updateWrapper.instance().props.history.location.pathname).toBe('/smsNotification');
    });
  });

  it('should call continueWithTip when clicked', () => {
    let props = {
      handlerForTipProp: jest.fn()
    };
    let continueWrapper = shallow(<TipCapture tipDetails={mockProps} isSmsEnabled {...props} nameCapture={false} deliveryDestination={false} history={history} subTotal={subTotal} tax={tax} customTipFlag currencyDetails={mockCurrency} selectedTipProp={'custom'} selectedTipAmountProp={1.00}/>);
    const spy = jest.spyOn(continueWrapper.instance(), 'continueWithTip');
    continueWrapper.instance().forceUpdate();
    continueWrapper.find('.pay-button').simulate('click');
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });

  it('should call continueWithoutTip', () => {
    let props = {
      handlerForTipProp: jest.fn()
    };
    let continueWrapper = shallow(<TipCapture tipDetails={mockProps} isSmsEnabled {...props}
      nameCapture={false} deliveryDestination={false} history={history} subTotal={subTotal} tax={tax}
      customTipFlag currencyDetails={mockCurrency} selectedTipProp={'custom'} selectedTipAmountProp/>);
    const spy = jest.spyOn(continueWrapper.instance(), 'continueWithoutTip');
    continueWrapper.instance().forceUpdate();
    continueWrapper.find('.no-tip-button').simulate('click');
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });
});
