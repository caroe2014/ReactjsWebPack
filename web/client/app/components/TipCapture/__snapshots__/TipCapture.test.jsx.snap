// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TipCapture does a snapshot test 1`] = `
<ThemeProvider
  className="tip-capture"
  theme={
    Object {
      "colors": Object {
        "addOnTab": "#347CFF",
        "addToCart": "#347CFF",
        "addonBg": "#EEEEEE",
        "backgroundGrey": "#E6E6E6",
        "black": "#000000",
        "close": "red",
        "contextBackground": "#f0f0f3",
        "darkGrey": "#464646",
        "disableBackground": "#E4E4E4",
        "disableButton": "#D1D1D1",
        "error": "#cc0000",
        "footer": "#48a942",
        "glutenFree": "#8d7356",
        "healthy": "#359919",
        "hint": "#a6a6a6",
        "inputBottomBorder": "lightgray",
        "itemBorder": "lightgrey",
        "itemTile": "#3C474C0",
        "light": "white",
        "loader": "#ababab",
        "loginBackground": "#303030",
        "loginBoxColor": "#434149",
        "loginField": "#AAA7AC",
        "loginFormBackground": "#484848",
        "navigationMenu": "#48a942",
        "navigationMenuHover": "cornflowerblue",
        "new": "#ff6000",
        "open": "#48a942",
        "paymentSuccessTitle": "#505050",
        "placeHolder": "#909090",
        "primary": "#347CFF",
        "primaryButton": "#347CFF",
        "primaryText": "#383838",
        "secondaryText": "#a6a6a6",
        "spicy": "#ed1c24",
        "text": "black",
        "textGrey": "#505050",
        "tile": "#505050",
        "validationError": "#cc0000",
        "vegan": "#0d4a22",
        "vegetarian": "#0d4a22",
      },
      "fontFamily": "Roboto",
      "fontSize": Object {
        "lg": "1.8rem",
        "md": "1.3rem",
        "nm": "1rem",
        "sm": "0.75rem",
        "xl": "2.4rem",
      },
      "fontSizeFromProps": [Function],
      "mediaBreakpoints": Object {
        "desktop": [Function],
        "mobile": [Function],
        "tablet": [Function],
      },
      "nav": Object {
        "height": "70px",
      },
      "receiptFontFamily": "Roboto",
      "shadow": "#cfcfcf 0.1px 0.1px 3px 0px",
      "spacing": Object {
        "lg": "4em",
        "md": "2em",
        "nm": "1em",
        "none": "0em",
        "sm": "0.5em",
        "xl": "8em",
      },
    }
  }
>
  <Styled(Flex)
    className="tip-capture-container"
  >
    <Styled(Flex)
      className="child-box"
    >
      <Styled(Flex)
        className="bill-box"
      >
        <Styled(Flex)
          className="bill-field sub-total"
          style={
            Object {
              "paddingBottom": "10px",
            }
          }
        >
          <Styled(Text)
            aria-label="TIP_CAPTURE_SUBTOTAL"
            className="bill-amount-label sub-total"
            tabIndex={0}
          >
            <div
              i18nKey="TIP_CAPTURE_SUBTOTAL"
            />
          </Styled(Text)>
          <Styled(Text)
            aria-label="$1.00"
            className="bill-amount-value sub-total"
            tabIndex={0}
          >
            $1.00
          </Styled(Text)>
        </Styled(Flex)>
        <Styled(Flex)
          className="bill-field"
        >
          <Styled(Text)
            aria-label="TIP_CAPTURE_TAX"
            className="bill-amount-label"
            tabIndex={0}
          >
            <div
              i18nKey="TIP_CAPTURE_TAX"
            />
          </Styled(Text)>
          <Styled(Text)
            aria-label="$1.00"
            className="bill-amount-value"
            tabIndex={0}
          >
            $1.00
          </Styled(Text)>
        </Styled(Flex)>
      </Styled(Flex)>
      <Styled(Flex)
        className="tip-box"
      >
        <Styled(Flex)
          className="bill-field bill"
        >
          <Styled(Text)
            aria-label="TIP_CAPTURE_TIP"
            className="add-tip-label"
            tabIndex={0}
          >
            <div
              i18nKey="TIP_CAPTURE_TIP"
            />
          </Styled(Text)>
          <Styled(Text)
            aria-label="$0.00"
            className="add-tip-label"
            tabIndex={0}
          >
            $0.00
          </Styled(Text)>
        </Styled(Flex)>
        <Styled(Flex)
          aria-label="TIP_SELECTION 3 OPTIONS_LABEL"
          className="tip-options-list"
          tabIndex={0}
        >
          <Styled(Flex)
            className="tip-button-0"
            onClick={[Function]}
            onKeyDown={[Function]}
            selected={false}
            tabIndex={0}
          >
            <Styled(Text)
              className="tip-option"
            >
              15
              %
            </Styled(Text)>
            <Styled(Text)
              className="tip-amount"
            >
              $0.15
            </Styled(Text)>
          </Styled(Flex)>
          <Styled(Flex)
            className="tip-button-1"
            onClick={[Function]}
            onKeyDown={[Function]}
            selected={false}
            tabIndex={0}
          >
            <Styled(Text)
              className="tip-option"
            >
              20
              %
            </Styled(Text)>
            <Styled(Text)
              className="tip-amount"
            >
              $0.20
            </Styled(Text)>
          </Styled(Flex)>
          <Styled(Flex)
            className="tip-button-2"
            onClick={[Function]}
            onKeyDown={[Function]}
            selected={false}
            tabIndex={0}
          >
            <Styled(Text)
              className="tip-option"
            >
              25
              %
            </Styled(Text)>
            <Styled(Text)
              className="tip-amount"
            >
              $0.25
            </Styled(Text)>
          </Styled(Flex)>
        </Styled(Flex)>
        <Styled(Flex)
          className="dynamic-boxes"
        >
          <Styled(Flex)
            className="no-tip-button"
            fullwidth={0}
            onClick={[Function]}
            tabIndex={0}
          >
            <div
              i18nKey="TIP_CAPTURE_NO_GRATUITY"
            />
          </Styled(Flex)>
          <Styled(Flex)
            className="custom-tip-button"
            onClick={[Function]}
            onKeyDown={[Function]}
            pointerclick={0}
            selected={false}
            tabIndex={0}
          >
            <Styled(Text)
              className="custom-tip-textss"
            >
              <div
                i18nKey="TIP_CAPTURE_CUSTOM"
              />
            </Styled(Text)>
          </Styled(Flex)>
        </Styled(Flex)>
      </Styled(Flex)>
      <Styled(Flex)
        className="button-box"
      >
        <Styled(Flex)
          className="pay-field"
        >
          <Styled(Text)
            aria-label="TIP_CAPTURE_AMOUNT_PAYABLE"
            className="bill-amount-label"
            tabIndex={0}
          >
            <div
              i18nKey="TIP_CAPTURE_AMOUNT_PAYABLE"
            />
          </Styled(Text)>
          <Styled(Text)
            aria-label="$1.00"
            className="bill-amount-value"
            tabIndex={0}
          >
            $1.00
          </Styled(Text)>
        </Styled(Flex)>
        <Styled(Flex)
          className="button-parent"
        >
          <Styled(Button)
            className="pay-button"
            onClick={[Function]}
            tabIndex={0}
          >
            <div
              i18nKey="DELIVERY_LOCATION_CONTINUE_PAY"
            />
          </Styled(Button)>
        </Styled(Flex)>
      </Styled(Flex)>
    </Styled(Flex)>
  </Styled(Flex)>
</ThemeProvider>
`;
